### PR TITLE
Add citation backlinks in chat response

### PR DESCRIPTION
**Pull request by chuyouchia**
Some basic citation backlinks in chat. On click it would bring you to the file itself
image

### CODE REVIEW

**Comment by samlhuillier** [comment]
A suggestion: it could be nicer to use a relative path here rather than full file path

**Comment by chuyouchia** [comment]
> A suggestion: it could be nicer to use a relative path here rather than full file path

image

**Comment by samlhuillier** [comment]
Also what do you think about not showing the file extension like we do in the file sidebar?

**Comment by chuyouchia** [comment]
> Also what do you think about not showing the file extension like we do in the file sidebar?

I thought that it was cleaner to keep the file extension, because we will probably move forward into support different types of files in the future and keeping the file extension allows us to preserve the navigation ability to those files (which is based on file path)

**Comment by chuyouchia** [comment]
> What do you think about the files referenced component being toggleable - hidden by default just below the assistant response?

Collapsed:
image

Expanded:
image

**Comment by samlhuillier** [comment]
I've just found out there is this: ```chunk.choices[0].finish_reason``` which I think we should use to detect when streaming has finished rather than managing that logic ourselves. What do you think?

**Review by samlhuillier — APPROVED** [review]
Looks good! (Removed the completed type too btw)

File: electron/main/database/dbSessionHandlers.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 import { getLLMConfig } from "../llm/llmConfig";
 import { errorToString } from "../Generic/error";
 import WindowsManager from "../windowManager";
 
+export interface PromptWithRagResults {
+  ragPrompt: string;
+  uniqueFilesReferenced: string[];
+}
+
+const MAX_COSINE_DISTANCE = 0.4;
```
[the remark this answers is not in the document]
Comment (chuyouchia) [reply electron/main/database/dbSessionHandlers.ts:16]: It was giving me really bad results without the bounds, and this ended up being good enough mostly

**Pull request merged**